OptimizelySDK/Odp/LruCache.cs

```diff
@@ -0,0 +1,171 @@
+﻿/* 
+ * Copyright 2022, Optimizely
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using OptimizelySDK.Logger;
+using OptimizelySDK.Utils;
+using System;
+using System.Collections.Generic;
+
+namespace OptimizelySDK.Odp
+{
+    public class LruCache<T> : ICache<T>
+        where T : class
+    {
+        private readonly int _maxSize;
+        private readonly object _mutex;
+        private readonly TimeSpan _timeout;
+        private readonly TimeSpan _timeoutDisabled = TimeSpan.Zero;
+        private readonly ILogger _logger;
+        private readonly Dictionary<string, (LinkedListNode<string> node, ItemWrapper value)> _cache;
+        private readonly LinkedList<string> _list;
+
+        public LruCache(int? maxSize = null,
```

Comment:
what should be the default? `null` is not default. Let's talk offline.

Reply:
Yes, let's talk.  I also need help with the default TimeSpan since it can't be a compile-time const.

Reply:
I've pushed a new solution for discussion.